src/components/browse/components/SelectedItemsController.js

```diff
@@ -0,0 +1,237 @@
+     * or columnDefinitions, which would allow us to put this below ColumnCombiner also
+     * if desired.
+     * Alternatively, attempt to detect based on presence of props.columnDefinitions
+     * or props.columnExtensionMap, throwing error if neither available.
+     */
+    columnExtensionMapWithSelectButton(){
+        const { columnExtensionMap: originalColExtMap, currentAction = null } = this.props;
+        const inSelectionMode = isSelectAction(currentAction);
+
+        if (!inSelectionMode || !originalColExtMap){
+            return originalColExtMap;
+        }
+
+        const columnExtensionMap = _.clone(originalColExtMap); // Avoid modifying in place
+
+        const origDisplayTitleRenderFxn = (
+            (originalColExtMap.display_title && originalColExtMap.display_title.render) ||
+            basicColumnExtensionMap.display_title.render
+        );
+
+        // Kept for reference in case we want to re-introduce constrain that for 'select' button(s) to be visible in search result rows, there must be parent window.
+        //var isThereParentWindow = inSelectionMode && typeof window !== 'undefined' && window.opener && window.opener.fourfront && window.opener !== window;
+
+        if (inSelectionMode) {
+            // Render out button and add to title render output for "Select" if we have a 'selection' currentAction.
+            // Also add the popLink/target=_blank functionality to links
+            // Remove lab.display_title and type columns on selection
+            columnExtensionMap.display_title = _.extend({}, columnExtensionMap.display_title, {
+                'minColumnWidth' : 120,
+                'render' : (result, columnDefinition, props, width) => {
+                    //set select click handler according to currentAction type (selection or multiselect)
+                    const { selectedItems } = this.state;
+                    const isChecked = selectedItems.has(itemUtil.atId(result));
+                    const isMultiSelect = (currentAction === 'multiselect');
+                    const checkBoxControl = (
+                        <input type="checkbox" checked={isChecked} onChange={this.handleSelectItemClick.bind(this, result, isMultiSelect)} className="mr-2" />
+                    );
+                    const currentTitleBlock = origDisplayTitleRenderFxn(
+                        result, columnDefinition, _.extend({}, props, { currentAction }), width, true
+                    );
+                    const newChildren = currentTitleBlock.props.children.slice(0);
+                    newChildren.unshift(checkBoxControl);
+                    return React.cloneElement(currentTitleBlock, { 'children' : newChildren });
+                }
+            });
+        }
+        return columnExtensionMap;
+    }
+
+    render(){
+        const { children, ...propsToPass } = this.props;
+        const { selectedItems } = this.state;
+        if (!React.isValidElement(children)){
```

Comment:
Will remove this as better/redundant one exists in the `React.Children.map(...` func below.